gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientReadChannel.java

```diff
@@ -494,6 +497,9 @@ private BlobSourceOption[] generateReadOptions(BlobId blobId) {
       if (blobId.getGeneration() != null) {
         readOptions.add(BlobSourceOption.generationMatch(blobId.getGeneration()));
       }
+      if (storageOptions.getEncryptionKey() != null) {
+        readOptions.add(BlobSourceOption.decryptionKey(storageOptions.getEncryptionKey().value()));
```

Comment:
I meant here. Here we are updating the readOptions, which is an input paramter. Changing the input parameter is usually a code smell.

Reply:
We are not mutating inputParameters but the local `readOptions` variable. Refactored the name to remove the confusion.